lib/_sendEvent.ts

```diff
@@ -1,66 +1,68 @@
-import { ClickReport } from './click';
-import { ConversionReport } from './conversion';
-import { SearchReport } from './search';
+import { InsightsSearchClickEvent } from './click';
+import { InsightsSearchConversionEvent } from './conversion';
 
-declare var process : {
+declare var process: {
   env: {
-    NODE_ENV: string
-  }
-}
+    NODE_ENV: string;
+  };
+};
 
-export type AnalyticsEvent = {
+export type InsightsEventTypes = 'click' | 'conversion';
+export type InsightsEvent = {
+  eventType: InsightsEventTypes;
+  eventName: string;
   userID: string;
   timestamp: number;
   indexName: string;
-}
-
-export type ReportEvent = 'search'|'click'|'conversion'
+};
 
 /**
  *  Sends data to endpoint
- * @param eventType 'click'|'conversion'
- * @param eventData ConversionReport|ClickReport
+ * @param eventType InsightsEventTypes
+ * @param eventData InsightsSearchClickEvent|InsightsSearchConversionEvent
  */
-export function sendEvent(eventType: ReportEvent, eventData: ClickReport | ConversionReport | SearchReport) {
-
+export function sendEvent(
+  eventType: InsightsEventTypes,
+  eventData: InsightsSearchClickEvent | InsightsSearchConversionEvent
+) {
   // Add client timestamp and userID
-  eventData.timestamp = Date.now()
+  eventData.timestamp = Date.now();
   eventData.userID = this._userID;
+  eventData.eventType = eventType;
+  // TODO: check eventType is matching eventData
 
-  // Origin
-  const reportingQueryOrigin = process.env.NODE_ENV === 'production' ? `https://insights.algolia.io/1/searches/${eventType}` : `http://localhost:8080/1/${eventType}`;
+  bulkSendEvent(this._applicationID, this.apiKey, [eventData]);
+}
 
+function bulkSendEvent(
+  applicationID: string,
+  apiKey: string,
+  events: InsightsEvent[]
+) {
+
+  const reportingQueryOrigin =
+    process.env.NODE_ENV === 'production' ? `https://insights.algolia.io/1/events` : `http://localhost:8080/1/events`;
```

Comment:
actually the solve purpose of this condition was testing.
I agree it doesn't belong inside prod code and should be removed.

It was not easy to remove because all the examples depend on it but I will in a next PR.
Created issue here: https://github.com/algolia/search-insights.js/issues/23